src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -391,10 +425,17 @@ struct
 
   let join a b =
     let join_d ad bd =
-      (*use copy of ad because result is later saved in there*)
-      let ad = Array.copy ad in
       (*This is the table which is later grouped*)
-      let table = BatList.map2i (fun i (ai, aj) (bi,bj) -> (i, Z.(aj - bj), (ai, aj), (bi,bj))) (Array.to_list ad) (Array.to_list bd) in
+      let joinfunction lhs rhs1 rhs2 = match rhs1, rhs2 with 
+        | Some (ai,aj),Some (bi,bj) -> Some (Z.(aj - bj),(ai,aj),(bi,bj))
+        | None, Some (bi,bj)        -> Some (Z.neg bj   ,Rhs.var_zero lhs,(bi,bj))
+        | Some (ai,aj), None        -> Some (aj         ,(ai,aj),Rhs.var_zero lhs)
```

Comment:
Could you elaborate again why this is not symmetric in the one is None case? Why does it matter which of the arguments the RHS comes from?

Reply:
I added specifics in the comments for this function now. If you still feel more need for explanation, you may reopen this issue of course :smile: